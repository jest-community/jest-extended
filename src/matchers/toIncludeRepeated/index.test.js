import matcher from './';

expect.extend(matcher);

describe('.toIncludeRepeated', () => {
  it('passes when string includes substring equal times', () => {
    expect('hello hello world').toIncludeRepeated('hello', 2);
  });

  it('fails when string includes substring less times', () => {
    expect(() => expect('hello world').toIncludeRepeated('hello', 2)).toThrowErrorMatchingSnapshot();
  });

  it('fails when string includes substring more times', () => {
    expect(() => expect('hello hello hello world').toIncludeRepeated('hello', 2)).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toIncludeRepeated', () => {
  it('passes when string includes substring less times', () => {
    expect('hello world').not.toIncludeRepeated('hello', 2);
  });

  it('passes when string includes substring more times', () => {
    expect('hello hello hello world').not.toIncludeRepeated('hello', 2);
  });

  it('fails when string includes substring equal times', () => {
    expect(() => expect('hello hello world').not.toIncludeRepeated('hello', 2)).toThrowErrorMatchingSnapshot();
  });
});
