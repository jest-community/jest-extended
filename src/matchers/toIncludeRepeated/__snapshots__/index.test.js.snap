// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toIncludeRepeated fails when string includes substring equal times 1`] = `
"<dim>expect(</><red>received</><dim>).not.toIncludeRepeated(</><green>expected</><dim>)</>

Expected string to not contain the following substring repeated 2 times: 
  <green>\\"hello\\"</>
Received: 
  <red>\\"hello hello world\\"</>"
`;

exports[`.toIncludeRepeated fails when string includes substring less times 1`] = `
"<dim>expect(</><red>received</><dim>).toIncludeRepeated(</><green>expected</><dim>)</>

Expected string to contain following substring repeated 2 times: 
  <green>\\"hello\\"</>
Received:
  <red>\\"hello world\\"</>"
`;

exports[`.toIncludeRepeated fails when string includes substring more times 1`] = `
"<dim>expect(</><red>received</><dim>).toIncludeRepeated(</><green>expected</><dim>)</>

Expected string to contain following substring repeated 2 times: 
  <green>\\"hello\\"</>
Received:
  <red>\\"hello hello hello world\\"</>"
`;
